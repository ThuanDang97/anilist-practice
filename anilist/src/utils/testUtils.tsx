import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => {
  return children
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export * from '@testing-library/user-event'
export { customRender as render }
