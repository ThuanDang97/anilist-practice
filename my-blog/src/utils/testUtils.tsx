import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import { RouterContext } from 'next/dist/shared/lib/router-context'

// Themes
import theme from '@self-types/Themes.types'
import { NextRouter } from 'next/router'

const createMockRouter = (router?: Partial<NextRouter>): NextRouter => ({
  basePath: '',
  pathname: '/',
  route: '/',
  query: {},
  asPath: '/',
  back: jest.fn(),
  beforePopState: jest.fn(),
  prefetch: jest.fn(),
  push: jest.fn(),
  reload: jest.fn(),
  replace: jest.fn(),
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  isFallback: false,
  isLocaleDomain: false,
  isReady: true,
  defaultLocale: 'en',
  domainLocales: [],
  isPreview: false,
  ...router,
})

export const customRender = ({
  children,
  router = { query: {} },
}: {
  children: React.ReactNode
  router: Partial<NextRouter>
}) => {
  return render(
    <ThemeProvider theme={theme}>
      <RouterContext.Provider value={createMockRouter(router)}>
        {children}
      </RouterContext.Provider>
    </ThemeProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
