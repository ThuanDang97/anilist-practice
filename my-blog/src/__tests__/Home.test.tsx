// Components
import Home from '@pages/index'

// Utils
import { render } from '@utils/testUtils'

describe('Welcome Component', () => {
  test('Component Welcome matches DOM Snapshot', () => {
    const { container } = render({ children: <Home />, router: {} })

    expect(container).toMatchSnapshot()
  })
})
