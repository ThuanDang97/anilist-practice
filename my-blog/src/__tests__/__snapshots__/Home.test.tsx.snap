// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component Component Welcome matches DOM Snapshot 1`] = `
<div>
  <div>
    Home Page
  </div>
</div>
`;
